### PR TITLE
virtual: expand APIDefinitionSetGetter

### PR DESCRIPTION
**Pull request by stevekuznetsov**
I saw in https://github.com/kcp-dev/kcp/pull/1052 @ncdc had similar requirements, so I'm pulling this out for us to share.

/assign @davidfestal @sttts @ncdc

### CODE REVIEW

**Comment by ncdc** [comment]
Is this to move the logical cluster from the APIDomainKey into the ctx?

**Comment by stevekuznetsov** [comment]
@ncdc no - I needed the context for a client call (but I guess I should be using a lister)

**Comment by stevekuznetsov** [comment]
@ncdc removed the ctx plumbing

File: pkg/virtual/framework/dynamic/apiserver/handler_test.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package apiserver
 
 import (
+	"context"
```
Comment (ncdc) [inline pkg/virtual/framework/dynamic/apiserver/handler_test.go:20]: Remove?

**Review by ncdc — APPROVED** [review]

**Pull request merged**